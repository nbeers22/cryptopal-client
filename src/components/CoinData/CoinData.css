.CoinData-hero{
  text-align: center;
}
.CoinData .description{
  width: 90%;
  max-width: 700px;
  margin: 0 auto;
  text-align: left;
}