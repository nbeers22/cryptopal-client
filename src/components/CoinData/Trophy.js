import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrophy } from '@fortawesome/free-solid-svg-icons'

const Trophy = props => {
  const { rank } = props
  let trophyColor
  switch (rank) {
    case 1:
      trophyColor = "gold"
      break;
    case 2:
      trophyColor = "silver"
      break;
    case 3:
      trophyColor = "bronze"
      break;
    default:
      trophyColor = undefined
  }
  return (
    <span className="Trophy">
      <FontAwesomeIcon
        icon={faTrophy}
        className={trophyColor}
      />
    </span>
  )
}

export default Trophy