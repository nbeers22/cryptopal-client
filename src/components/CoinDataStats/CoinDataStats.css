.CoinData-stats{
  background-color: #fff;
}
.CoinData-stats .stats-table{
  display: flex;
}
.CoinData-stats aside{
  flex-grow: 1;
  text-align: center;
}

@media(max-width: 767px){
  .CoinData-stats .stats-table{
    flex-wrap: wrap;
  }
  .CoinData-stats aside{
    flex: 0 0 50%;
    margin-bottom: 30px;
  }
}