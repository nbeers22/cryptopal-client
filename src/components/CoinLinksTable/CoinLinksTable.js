import React from 'react'

const CoinLinksTable = props => {
  const { urls, name } = props

  const removeSlashes = url => {
    return url.replace(/^(?:https?:\/\/)?(?:www\.)?/i, "").split("/")[0]
  }

  return(
    <aside className="coin-links">
      <table>
        <thead>
          <tr>
            <th colSpan="2">
              <h2>{ `${name} Resources` }</h2>
            </th>
          </tr>
        </thead>
        <tbody>

          { urls.website[0] &&
            <tr>
              <td>Website</td>
              <td><a className="btn-cta" href={urls.website[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.website[0])}</a></td>
            </tr>
          }

          { urls.announcement[0] &&
              <tr>
                <td>Announcements</td>
                <td><a className="btn-cta" href={urls.announcement[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.announcement[0])}</a></td>
              </tr>
          }
          
          { urls.chat[0] &&
            <tr>
              <td>Chat</td>
              <td><a className="btn-cta" href={urls.chat[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.chat[0])}</a></td>
            </tr>
          }

          { urls.explorer[0] &&
            <tr>
              <td>Explorer</td>
              <td><a className="btn-cta" href={urls.explorer[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.explorer[0])}</a></td>
            </tr>
          }

          { urls.message_board[0] &&
            <tr>
              <td>Message Board</td>
              <td><a className="btn-cta" href={urls.message_board[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.message_board[0])}</a></td>
            </tr>
          }

          { urls.source_code[0] &&
            <tr>
              <td>Source Code</td>
              <td><a className="btn-cta" href={urls.source_code[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.source_code[0])}</a></td>
            </tr>
          }
          
          { urls.technical_doc[0] &&
            <tr>
              <td>Technical Doc</td>
              <td><a className="btn-cta" href={urls.technical_doc[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.technical_doc[0])}</a></td>
            </tr>
          }

          { urls.reddit[0] &&
            <tr>
              <td>Reddit</td>
              <td><a className="btn-cta" href={urls.reddit[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.reddit[0])}</a></td>
            </tr>
          }

          { urls.twitter[0] &&
            <tr>
              <td>Twitter</td>
              <td><a className="btn-cta" href={urls.twitter[0]} target="_blank" rel="noopener noreferrer">{removeSlashes(urls.twitter[0])}</a></td>
            </tr>
          }

        </tbody>
      </table>
    </aside>
  )
}

export default CoinLinksTable