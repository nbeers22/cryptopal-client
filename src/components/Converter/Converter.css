.Converter{
  background-color: #CCC;
}
.Converter form{
  display: flex;
  margin: 0 auto;
  max-width: 500px;
  justify-content: center;
}
.Converter-block{
  margin: 10px;
  display: flex;
}
.Converter-block label{
  background-color: #EEE;
  padding: 10px;
}
.Converter-block input{
  background-color: #FFF;
  padding: 5px 10px;
  border: 0;
  font-size: 16px;
  -moz-appearance:textfield;
}
.Converter-block input::-webkit-outer-spin-button,
.Converter-block input::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
}
.Converter-block svg{
  font-size: 2em;
}
@media(min-width: 641px){
  .Converter-block .show-mobile{
    display: none;
  }
}

@media(max-width: 640px){
  .Converter form{
    flex-wrap: wrap;
  }
  .Converter-block{
    flex: 0 0 100%;
    justify-content: center;
  }
  .Converter-block .hide-mobile{
    display: none;
  }
}