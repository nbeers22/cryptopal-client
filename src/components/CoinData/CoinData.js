import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faReddit, faTwitter } from '@fortawesome/free-brands-svg-icons'
import config from '../../config.js'
import './CoinData.css'

class CoinData extends Component {
  state = {
    coinData: {
      1: {
        name: "",
        urls: {
          website: "",
          technical_doc: "",
          twitter: "",
          reddit: "",
          message_board: "",
          explorer: [],
          source_code: ""
        },
        logo: "",
        symbol: "",
        description: "",
        tags: []
      }
    }
  }
  
  componentDidMount(){
    const coinID = this.props.match.params.coin_id;
    const url = `${config.API_URL}/coins/${coinID}`;
    fetch(url)
    .then( response => {
      return response.ok
        ? response.json()
        : console.log(response.statusText)
    })
    .then( responseJson => {
      this.setState({
        coinData: responseJson.data
      })
    })
    .catch( error => console.log(error))
  }

  render() {
    const { name, urls, logo, symbol, description, tags  } = this.state.coinData[1];

    return (
      <section className="CoinData">
        <div className="container">
          <section className="CoinData-hero">
            <img src={logo} alt=""/>
            <h1>{name}</h1>
            <p className="description">{description}</p>
          </section>
          <aside className="coin-links">
            <h3>Links</h3>
            <p><a href={urls.website} target="_blank">Website</a></p>
            <ul>
              <li><a href={urls.reddit} target="_blank"><FontAwesomeIcon icon={faReddit} /></a></li>
              <li><a href={urls.reddit} target="_blank"><FontAwesomeIcon icon={faTwitter} /></a></li>
            </ul>
            <p><a href={urls.website}>Website</a></p>
            <p><a href={urls.website}>Website</a></p>
            <p><a href={urls.website}>Website</a></p>
          </aside>
        </div>
      </section>
    )
  }
}

export default withRouter(CoinData);