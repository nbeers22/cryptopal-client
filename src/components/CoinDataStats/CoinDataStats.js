import React, { Component } from 'react'
import './CoinDataStats.css'

export default class CoinDataStats extends Component {
  state = {
    market_cap: 0,
    percent_change_1h: 0,
    percent_change_24h: 0,
    percent_change_7d: 0,
    volume_24h: 0
  }

  componentWillReceiveProps(props){
    this.setState({ ...props.prices })
  }

  currencyFormat(num){
    return num && '$' + num.toFixed(2).replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1,')
  }

  percentClass(percent){
    return percent > 0 ? 'green' : 'red';
  }

  render() {
    const { 
      market_cap,
      percent_change_1h,
      percent_change_24h,
      percent_change_7d,
      volume_24h
    } = this.state

    return (
      <section className="CoinData-stats">
      <div className="container">
        <div className="stats-table">
          <aside>
            <p><strong>Market Cap</strong></p>
            { this.currencyFormat(market_cap) }
          </aside>
          <aside>
            <p><strong>Change 1hr</strong></p>
            <span className={this.percentClass(percent_change_1h)}>{ percent_change_1h.toFixed(2) + '%' }</span>
          </aside>
          <aside>
            <p><strong>Change 24hr</strong></p>
            <span className={this.percentClass(percent_change_24h)}>{ percent_change_24h.toFixed(2) + '%' }</span>
          </aside>
          <aside>
            <p><strong>Change 7d</strong></p>
            <span className={this.percentClass(percent_change_7d)}>{ percent_change_7d.toFixed(2) + '%' }</span>
          </aside>
          <aside>
            <p><strong>Volume 24hr</strong></p>
            { this.currencyFormat(volume_24h) }
          </aside>
        </div>
      </div>
    </section>
    )
  }
}
