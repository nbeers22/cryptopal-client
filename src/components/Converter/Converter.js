import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faArrowsAltH, faArrowsAltV } from '@fortawesome/free-solid-svg-icons'
import './Converter.css'

export default class Converter extends Component {
  
  constructor(props){
    super(props);
    this.state = {
      symbol: '',
      valueUSD: null
    }
  }

  componentWillReceiveProps(props){
    const { symbol, valueUSD } = props

    this.setState({
      symbol,
      valueUSD
    })
  }

  convertToUSD = event => {
    const { value } = event.target;
    const convertedValue = value * this.state.valueUSD;
    const totalUSD = value ? convertedValue.toFixed(2) : ""

    document.getElementById('currency-input').value = totalUSD;
  }
  
  convertToCoin = event => {
    const { value } = event.target;
    const convertedValue = value / this.state.valueUSD;
    const totalCoins = value ? convertedValue : ""

    document.getElementById('coin-input').value = totalCoins;
  }

  render() {
    const { symbol } = this.state;

    return (
      <section className="Converter">
        <form>
          <div className="Converter-block">
            <label htmlFor="coin-input">{ symbol }</label>
            <input id="coin-input" type="number" onChange={this.convertToUSD} />
          </div>
          <div className="Converter-block" style={{ alignItems: "center" }}>
            <span className="hide-mobile"><FontAwesomeIcon icon={faArrowsAltH} /></span>
            <span className="show-mobile"><FontAwesomeIcon icon={faArrowsAltV} /></span>
          </div>
          <div className="Converter-block">
            <label htmlFor="currency-input">USD</label>
            <input id="currency-input" type="number" onChange={this.convertToCoin} />
          </div>
        </form>
      </section>
    )
  }
}
