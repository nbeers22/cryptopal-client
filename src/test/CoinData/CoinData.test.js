import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter as Router } from 'react-router-dom'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import renderer from 'react-test-renderer';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faReddit } from '@fortawesome/free-brands-svg-icons'
import PriceGraph from '../../components/PriceGraph/PriceGraph.js'
import Converter from '../../components/Converter/Converter.js'
import RedditFeedEntry from '../../components/RedditFeedEntry/RedditFeedEntry.js';
import CoinLinksTable from '../../components/CoinLinksTable/CoinLinksTable.js';
import CoinDataStats from '../../components/CoinDataStats/CoinDataStats.js';
import config from '../../config.js'
import loader from '../../images/loading.gif'
import Trophy from '../../components/CoinData/Trophy.js';
import CoinData from '../../components/CoinData/CoinData.js'

describe('CoinData Component',  () => {
  let wrapper,fetchSpy
  beforeEach( async () => {
      fetchSpy = jest.spyOn(window, 'fetch');
      wrapper = await shallow(
        <CoinData.WrappedComponent 
          match={{ params: { coin_id: 1 } }}
        />
      )
  })

  it("should render content if initial data is available", () => {
    expect(fetchSpy).toBeCalled();
    // console.log(fetchSpy)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  
  // it('renders without errors', async () => {
    // const instance = wrapper.instance();
    // Now we call the componentDidMount event, telling the component that it mounted. But because
    // we called it manually we are able to await for it to resolve. This makes sure the promise
    // for the method is completed before going on with the code.
    // await instance.componentDidMount();
  // })

})
