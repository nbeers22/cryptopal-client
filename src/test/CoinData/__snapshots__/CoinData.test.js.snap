// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinData Component should render content if initial data is available 1`] = `
<div
  className="CoinData"
>
  <section
    className="CoinData-hero"
  >
    <div
      className="container"
    >
      <img
        alt=""
        src=""
      />
      <h1>
         ()
      </h1>
      <p>
        <span
          className="rank"
        >
          <Trophy
            rank={null}
          />
           
          Rank #null
        </span>
      </p>
      <h2 />
      <p
        className="description"
      />
    </div>
  </section>
  <CoinDataStats
    prices={
      Object {
        "market_cap": null,
        "percent_change_1h": null,
        "percent_change_24h": null,
        "percent_change_7d": null,
        "price": null,
        "volume_24h": null,
      }
    }
  />
  <PriceGraph
    slug=""
  />
  <Converter
    symbol=""
    valueUSD={null}
  />
  <section
    className="CoinData-info"
  >
    <div
      className="container"
    >
      <CoinLinksTable
        name=""
        urls={
          Object {
            "announcement": Array [],
            "chat": Array [],
            "explorer": Array [],
            "message_board": "",
            "reddit": "",
            "source_code": "",
            "technical_doc": "",
            "twitter": "",
            "website": "",
          }
        }
      />
      <aside
        className="reddit-feed-container"
      >
        <img
          alt="Loading Reddit Feed"
          src="loading.gif"
        />
      </aside>
    </div>
  </section>
</div>
`;
